agent/utils/ephemeral_ports.go

```diff
@@ -58,3 +63,82 @@ func GenerateEphemeralPortNumbers(n int, reserved []uint16) ([]uint16, error) {
 	}
 	return result, nil
 }
+
+var dynamicHostPortRange = getDynamicHostPortRange
+
+// GetHostPortRange gets a set of contiguous host ports from the ephemeral host port range defined on the host. The
+// number of host ports is determined by the number of container ports, in the given containerPortRange.
+func GetHostPortRange(containerPortRange string, protocol string) (string, error) {
+	numberOfPorts, err := getPortRangeLength(containerPortRange)
+	if err != nil {
+		return "", err
+	}
+
+	// get ephemeral port range, either default or if custom-defined
+	startHostPortRange, endHostPortRange, err := dynamicHostPortRange()
+	if err != nil {
+		seelog.Warnf("Unable to read the ephemeral host port range, falling back to the default range: %v-%v",
+			defaultPortRangeStart, defaultPortRangeEnd)
+		startHostPortRange, endHostPortRange = defaultPortRangeStart, defaultPortRangeEnd
+	}
+
+	var startPort, lastPort, n int
+	for port := startHostPortRange; port <= endHostPortRange; port++ {
```

Comment:
Should we put some kind of synchronization mechanism around this block to avoid race condition when two tasks simultaneously need host port range allocation?

Reply:
Ideally any synchronization should be happening in the task launch workflow, and not in utility functions such as this. Are there are no other host resource races that the agent has to guard against?

Reply:
agreed. Yinyi, Yash and I did discussed this earlier today. The upcoming refactor creates a port mapping manager that handles synchronization and tracks the last port agent assigned to a container. 

to follow soon.

Reply:
discussed offline. We're going to add a lock here. Next, I am going to follow-up with a revised design aiming to improve on synchronization and port tracking.